ui/imports/shared/status/StatusChatInput.qml

```diff
@@ -1260,13 +1261,13 @@ Rectangle {
                             }
 
                             onCursorPositionChanged: {
-                                if(mentionsPos.length > 0) {
+                                if(mentionsPos.length > 0 && ((keyEvent.key === Qt.Key_Left) || (keyEvent.key === Qt.Key_Right))) {
```

Comment:
WDYT of checking if the cursor changed due to text length change?
I think keyEvent property is not reliable here because the cursor can also change with mouse input. Also, checking for left and right key might not be enough because the cursor can be moved also with up/down and probably other standard shortcuts.

Reply:
this was exactly part of the problem here,  it was getting inside here in all cases and thus when the user was typing anything before the mention it made the cursor jumping to the end of the mention and strange characters appeared. What we only need to check here is if the user pressed right or left arrow key after or before a mention respectively so that we can "jump" over it. let me know if there's any other use case I'm missing!

Reply:
https://user-images.githubusercontent.com/47811206/213663851-4ecbc3ee-58fd-4839-8736-dcaf6da5a6c7.mov

Here is an example of why  && ((keyEvent.key === Qt.Key_Left) || (keyEvent.key === Qt.Key_Right))) is not enough.
Now we can move the cursor with mouse, land inside a mention and alter it. The result will be the same - invalid mention.

It's a tricky thing to fix and also preserve jumping over mentions. Will look into some options. Please let me know if you  an idea on this.

Reply:
@alexjba I have updated it and it works when it comes to selections. the question is now what to do when we have selected text and we type another character, discord for example removes all selected characters and replaces with the new one. @benjthayer any thoughts here?


https://user-images.githubusercontent.com/31625338/213711102-d8d37398-4c04-4cc4-9919-9e88cabfc481.mov

Reply:
Ohh, looks like it's from this:
StatusChatInput.qml:369 event.accepted = true
StatusChatInput.qml:373 event.accepted = true

It breaks the default behaviour :(

Reply:
food catch! pushed, would you mind checking again?

Reply:
> @alexjba I have updated it and it works when it comes to selections. the question is now what to do when we have selected text and we type another character, discord for example removes all selected characters and replaces with the new one. @benjthayer any thoughts here?
> ta.mov

@benjthayer please ignore my comment here, it was a bug in code!

Reply:
> the question is now what to do when we have selected text and we type another character, discord for example removes all selected characters and replaces with the new one. @benjthayer any thoughts here?

@alexandraB99 yes, I think any selected text should be replaced by any newly typed text - that includes mentions. Mentions are a single entity so the user shouldn't be able to highlight/select a fragment of it. 

The user should also be able to position the cursor to directly before / directly after a mention and use SHIFT+RIGHT / SHIFT+LEFT which will select the entire mention - so the user can delete/overtype it.

Reply:
Ah! Replied at the same time! No worries - probably just the SHIFT key paragraph that's relevant then :)

Reply:
@benjthayer great, created a new issue for this https://github.com/status-im/status-desktop/issues/9216. just to confirm, when user selects some text and types new character(s), the selected text should be replaced with the new characters independently if it includes mentions or not, right?

Reply:
@alexandraB99 yes - just like Discord. If I have selected/highlighted text - it doesnt matter if that selected/highlighted text includes a mention,  it should all be replaced with the text that is typed while that other text is selected/highlighted.

And you should not be able to highlight only part of a mention - a mention is either entirely highlighted or entirely unhighlighted. If it is highlighted and the user types, it is replaced by the new text.